test/extensions/filters/http/oauth2/filter_test.cc

```diff
@@ -155,6 +157,82 @@ class OAuth2Test : public testing::Test {
   Event::SimulatedTimeSystem test_time_;
 };
 
+// Verifies that the OAuth SDSSecretReader correctly updates dynamic generic secret
+TEST_F(OAuth2Test, SdsDynamicGenericSecret) {
+  NiceMock<Server::MockConfigTracker> config_tracker;
+  std::unique_ptr<Secret::SecretManager> secret_manager(
+      new Secret::SecretManagerImpl(config_tracker));
```

Comment:
You can probably get by here by allocating this on the stack?

Reply:
Done